src/index.ts

```diff
@@ -385,18 +395,66 @@ export async function cli(args: string[]) {
     namedExports: undefined,
     webDependencies: undefined,
   };
-  const doesWhitelistExist = !!webDependencies;
-  const arrayOfDeps = webDependencies || Object.keys(pkgManifest.dependencies || {});
+  let doesWhitelistExist = !!webDependencies;
+  let arrayOfDeps = webDependencies || Object.keys(pkgManifest.dependencies || {});
   const hasBrowserlistConfig =
     !!pkgManifest.browserslist ||
     !!process.env.BROWSERSLIST ||
     fs.existsSync(path.join(cwd, '.browserslistrc')) ||
     fs.existsSync(path.join(cwd, 'browserslist'));
+  let dependencyTree: string | undefined;
+
+  // auto detection with --entry flag
+  if (typeof entry === 'string' && entry.length) {
+    const files = entry.split(',').map(file => path.resolve(cwd, file));
+    const fileDeps = autoResolve(files, cwd);
+
+    // print out dependency tree if specified
+    if (depTree) {
+      dependencyTree = JSON.stringify({dependencies: fileDeps});
+    }
+
+    // this will examine package.json dependencies for any matches.
+    const destRoot = dest.replace(/\$/, '').split('/'); // take last segment of --dest
+    if (pkgManifest.dependencies) {
+      function npmName(filename: string) {
+        const npmRoot = filename.split(`${destRoot[destRoot.length - 1]}/`)[1]; // determine npm root based on --dest
+        if (!npmRoot) {
+          return;
+        }
+        const moduleName = npmRoot.replace(/\.[A-z]+$/, ''); // remove .js extension
+        return moduleName.startsWith('@')
+          ? moduleName.replace(/(\/[^/]*).*/, '$1')
+          : moduleName.replace(/\/.*/, ''); // keep first slash if scoped, or remove all slashes if non-scoped
+      }
+
+      // for each dependency, add it if it’s new
+      Object.values(fileDeps).forEach(depList => {
+        depList.forEach(dep => {
+          const moduleName = npmName(dep);
+          if (pkgManifest.dependencies[moduleName] && !arrayOfDeps.includes(moduleName)) {
+            arrayOfDeps.push(moduleName);
+          }
+        });
+      });
+    } else {
+      console.warn(chalk.yellow('No dependencies found in package.json to resolve'));
+    }
```

Comment:
This is the gross bit that’s a little brittle. Let me explain what some of this is doing.

In my sample project, I’m using ZEIT Now. That means my build directory is `/public/`. So to @pika/pack, I use `pika-pack --dest public/web_modules`. Good so far.

However, in my JS, I’m not specifying `/public/web_modules/vue.js`; I’m still using `/web_modules/vue.js`. So how do I intelligently tell what’s an npm package and what isn’t?

This script grabs the last segment of `--dest`, which is `web_modules` in this case. And it does a little bit of juggling to see if you take away the `--dest` top folder if it matches something in `dependencies` (it also has to worry about whether it’s a scoped package or not). 

This whole block here that figures out that `/web_modules/vue/dist/vue.esm.browser.js` is the same thing as `vue` in `dependencies` could definitely use some polish.

Reply:
Ah, gotcha. I should have read this before commenting :)

a few things:
1. some projects (like www.pika.dev) are importing by package name, and then using the built-in Babel plugin to rewrite to `/web_modules/package-name.js` on build. This should be able to support that usecase.
2. Let's aim for simple/explicit to start, and then we can get smarter once this has baked a bit. I'd recommend: scan for bare package names, or scan for `${dest}/*.js`, OR allow the user to specify a special prefix via config (importDest? importPath?) and then look for `${importPath}/*.js`.

Does that make sense?

Reply:
For #1, because this runs before the Babel step, I don’t think it applies? Unless I’m misunderstanding.

For #2, I actually started that way, but hesitated adding another flag. That is simpler, though, you’re right. But I’m also torn because `--dest` kinda _is_ related to all these imports in some way, so I wanted to use it somehow. I have a feeling you’re onto something, though, and simplifying things and adding a flag is fine with me as a first pass.

Reply:
Resolving this convo for cleanup; I’ll ask for some missing clarity in another comment.